src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -84,6 +96,25 @@ void FlightTaskManualAltitude::_scaleSticks()
 	// scale horizontal velocity with expo curve stick input
 	const float vel_max_z = (_sticks(2) > 0.0f) ? _constraints.speed_down : _constraints.speed_up;
 	_velocity_setpoint(2) = vel_max_z * _sticks_expo(2);
+
+	// if there is a valid distance to bottom or distance to home, then
+	// adjust speed downwards gradually within the limits MPC_LAND_ALT1 and MPC_LAND_ALT2.
+	if (_sticks(2) > 0.0f) { //user demands speed downwards
+
+		float dist_to_ground = NAN;
+
+		if (PX4_ISFINITE(_dist_to_bottom)) {
+			dist_to_ground = _dist_to_bottom;
+
+		} else if (PX4_ISFINITE(_sub_home_position->get().valid_alt)) {
```

Comment:
it will slow-down as well. This "feature" is actually just a legacy "feature" (https://github.com/PX4/Firmware/pull/9368/files#diff-fc77c3ef569029d45764664c75d4b0c1L2429), I just added it back.

Unfortunately, I cannot find anymore the discussion we had when this "feature" got introduced. I was strongly against it, but the decision was different. At least this time I added the check that if there is a distance sensor to ground, then use the distance to ground instead of blindly use the altitude above home